### PR TITLE
Added getNeighbors() and registerClicks() to lights-out.js

### PR DESCRIPTION
**Pull request by Cameron-Taleghani**
The function `getNeighbors()` returns an array of all the neighbors to be toggled of the current button clicked. The function `registerClick()` registers when a button is clicked and toggles the button and its neighbors.

### CODE REVIEW

File: lights-out.js [hunk 1]
```diff
@@ -16,4 +17,51 @@ class LightsOut {
             }
         }
     }
+
+    getNeighbors(row, col) {
```
Comment (mikegagnon) [inline lights-out.js:21]: please change this to a static method

File: lights-out.js [hunk 2]
```diff
@@ -16,4 +17,51 @@ class LightsOut {
...
+        const neighborArray = [];
+        
+        if(row < this.numRows - 1) {
+            if(row > 0) {
+                neighborArray.push([row + 1, col]);
+                neighborArray.push([row - 1, col]);
+            } else {
+                neighborArray.push([row + 1, col]);
+                neighborArray.push([this.numRows - 1, col]);
+            }
```
Comment (mikegagnon) [inline lights-out.js]: the push `neighborArray.push([row + 1, col])` occurs the same regardless of the conditional (i.e. if (row > 0)

Therefore, you should move that push to above the conditional.

File: lights-out.js [hunk 3]
```diff
@@ -16,4 +17,51 @@ class LightsOut {
             }
         }
     }
+
+    getNeighbors(row, col) {
+        const neighborArray = [];
+        
+        if(row < this.numRows - 1) {
```
Comment (mikegagnon) [inline lights-out.js]: I notice there is no missing whitespace between `if` and `(row ...)`. Please insert spaces throughout to achieve consistency in your PR.

Also, I noticed the same issue with missing whitespace in other areas tangential to this PR. Do not fix those whitespace issues in this PR, but after I merge this PR, please submit a new PR fixing whitespace.

File: lights-out.js [hunk 4]
```diff
@@ -16,4 +17,51 @@ class LightsOut {
...
+                neighborArray.push([row + 1, col]);
+                neighborArray.push([this.numRows - 1, col]);
+            }
+        } else {
+            neighborArray.push([0, col]);
+            neighborArray.push([row - 1, col]);
+        }
+        if(col < this.numCols - 1) {
+            if(col > 0) {
+                neighborArray.push([row, col + 1]);
```
Comment (mikegagnon) [inline lights-out.js]: Similar to my previous comment, the line `neighborArray.push([row, col + 1]);` appears twice. Delete those lines please, and move the line to above the `if (col >0)` conditional

File: lights-out.js [hunk 5]
```diff
@@ -16,4 +17,51 @@ class LightsOut {
...
+            } else {
+                neighborArray.push([row, col + 1]);
+                neighborArray.push([row, this.numCols - 1]);     
+            }
+        } else {
+            neighborArray.push([row, 0]);
+            neighborArray.push([row, col - 1]);
+        }
+
+        console.log(neighborArray);
```
Comment (mikegagnon) [inline lights-out.js]: Please remove all console logs.

File: lights-out.js [hunk 6]
```diff
@@ -16,4 +17,51 @@ class LightsOut {
...
+
+    registerClicks() {
+        for(let row = 0; row < this.numRows; row++) {
+            for(let col = 0; col < this.numCols; col++) {
+                const button = document.querySelector(`#${this.gameId} [data-row="${row}"][data-col="${col}"]`);
+                button.addEventListener('click', () => {
+                    button.classList.toggle("on");
+                    const neighbors = this.getNeighbors(row, col);
+                    
+                    neighbors.forEach(neighbor => document.querySelector(`#${this.gameId} [data-row="${neighbor[0]}"][data-col="${neighbor[1]}"]`)
```
Comment (mikegagnon) [inline lights-out.js]: I think the use of `forEach` in this context is a little confusing. I suggest instead:

```
for (const neighbor of neighbors) {
    const [r, c] = neighbor;
    document.querySelector(`#${this.gameId} [data-row="${r}"][data-col="${c}"]`)
        .classList.toggle("on")
}
```

**Review by mikegagnon — CHANGES_REQUESTED** [review]
Great work! Just a few comments.

File: lights-out.js (hunk 3 above)
Comment (Cameron-Taleghani) [reply lights-out.js]: Added whitespace after the `if` statements in `getNeighbors()`. I notice the same issue at other `if` statements and `for` loops. I assume these are the ones you want me to fix in another PR, therefore I will leave them for now.

File: lights-out.js (hunk 4 above)
Comment (Cameron-Taleghani) [reply lights-out.js]: Moved `neighborArray.push([row, col + 1]);` to above the `if (col >0)` conditional.

File: lights-out.js (hunk 5 above)
Comment (Cameron-Taleghani) [reply lights-out.js]: Removed console log on line `49`.

File: lights-out.js (hunk 6 above)
Comment (Cameron-Taleghani) [reply lights-out.js]: Added 

```
for (const neighbor of neighbors) {
    const [r, c] = neighbor;
    document.querySelector(`#${this.gameId} [data-row="${r}"][data-col="${c}"]\`)
        .classList.toggle("on")
}
```

to `registerClicks()` and removed 

```
neighbors.forEach(neighbor => document.querySelector(`#${this.gameId} [data-row="${neighbor[0]}"][data-col="${neighbor[1]}"]`)`
```

File: lights-out.js (hunk 1 above)
Comment (mikegagnon) [reply lights-out.js:21]: oops. `getNeighbors` should *not* be static. sorry about that.

**Review by mikegagnon — APPROVED** [review]
Great, thanks!

**Pull request merged**